src/braket/emulation/device_emulator_properties.py

```diff
@@ -0,0 +1,216 @@
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+import json
+
+from braket.device_schema.device_capabilities import DeviceCapabilities
+from braket.device_schema.ionq.ionq_device_capabilities_v1 import IonqDeviceCapabilities
+from braket.device_schema.result_type import ResultType
+from braket.device_schema.standardized_gate_model_qpu_device_properties_v1 import (
+    OneQubitProperties,
+    TwoQubitProperties,
+)
+from braket.schema_common.schema_base import BraketSchemaBase
+
+from braket.circuits.translations import BRAKET_GATES
+from braket.emulation._standardization import _standardize_ionq_device_properties
+
+
+class DeviceEmulatorProperties:
+    """Properties for device emulation.
+
+    Args:
+        qubitCount (int): Number of qubits in the device
+        nativeGateSet (list[str]): List of native gates supported by the device. Must be valid
+            Braket gates. Valid gates include: gphase, i, h, x, y, z, cv, cnot, cy, cz, ecr, s, si,
+            t, ti, v, vi, phaseshift, cphaseshift, cphaseshift00, cphaseshift01, cphaseshift10, rx,
+            ry, rz, U, swap, iswap, pswap, xy, xx, yy, zz, ccnot, cswap, gpi, gpi2, prx, ms, unitary
+        connectivityGraph (dict[str, list[str]]): Graph representing qubit connectivity. If it is an
+            empty dictionary, the device is treated as fully connected.
+        oneQubitProperties (dict[str, OneQubitProperties]): Properties of one-qubit calibration
+            details
+        twoQubitProperties (dict[str, TwoQubitProperties]): Properties of two-qubit calibration
+            details
+        supportedResultTypes (list[ResultType]): List of supported result types.
+    """
+
+    def __init__(
+        self,
+        qubitCount: int,
+        nativeGateSet: list[str],
+        connectivityGraph: dict[str, list[str]],
+        oneQubitProperties: dict[str, OneQubitProperties],
+        twoQubitProperties: dict[str, TwoQubitProperties],
+        supportedResultTypes: list[ResultType],
+    ):
+        """Initialize a DeviceEmulatorProperties instance."""
+        # Validate inputs
+        self.validate_nativeGateSet(nativeGateSet)
+        self.validate_oneQubitProperties(oneQubitProperties, qubitCount)
+
+        # Get qubit labels for further validation
+        indices = list(oneQubitProperties.keys())
+        qubit_labels = sorted(int(x) for x in indices)
+
+        self.validate_connectivityGraph(connectivityGraph, qubit_labels)
+        self.validate_twoQubitProperties(twoQubitProperties, qubit_labels)
+
+        # Store properties
+        self.qubitCount = qubitCount
+        self.nativeGateSet = nativeGateSet
+        self.connectivityGraph = connectivityGraph
+        self.oneQubitProperties = oneQubitProperties
+        self.twoQubitProperties = twoQubitProperties
+        self.supportedResultTypes = supportedResultTypes
```

Comment:
Can you make these snake_case, and make them properties (and the attributes themselves private)?

Reply:
done